api/grpc/mpi/v1/file.proto

```diff
@@ -0,0 +1,133 @@
+
+syntax = "proto3";
+package mpi.v1;
+
+option go_package = "github.com/nginx/agent/v3/api/grpc/mpi/v1";
+
+import "mpi/v1/common.proto";
+import "google/protobuf/timestamp.proto";
+
+// This specifies the FileService operations for transferring file data between a client and server.
+// All operations are written from a client perspective and flow Client -> Server
+// The server must set a max file size (in bytes), and that size must be used to configure 
+// the gRPC server and client for the FileService such that the FileContents object can be sent with bytes of the configured size. 
+// The actual configured max size for gRPC objects must be maxFileSize + sizeOfSha256HashString since a FileContents object contains both. 
+// A SHA256 hash string is 64 bytes, therefore the configured max message size should be maxFileSize + 64.
+service FileService {
+    // Get the overview of files for a particular configuration version of an instance
+    rpc GetOverview(GetOverviewRequest) returns (GetOverviewResponse) {}
+    // Update the overview of files for a particular set of file changes on the data plane
+    rpc UpdateOverview(UpdateOverviewRequest) returns (UpdateOverviewResponse) {}
+    // Get the file contents for a particular file
+    rpc GetFile(GetFileRequest) returns (GetFileResponse) {}
+    // Update a file from the Agent to the Server
+    rpc UpdateFile(UpdateFileRequest) returns (UpdateFileResponse) {}
+}
+
+// Represents a request payload for a file overview
+message GetOverviewRequest {
+    // The config version of the overview you are requesting
+    ConfigVersion config_version = 1;
+}
+
+// Represents a response payload for an overview of files for a particular configuration version of an instance
+message GetOverviewResponse {
+    // The file overview of 
+    FileOverview overview = 1;
+}
+
+// Represents a the payload for an overview an update of  files for a particular configuration version of an instance
+message UpdateOverviewRequest {
+    // The file overview of 
+    FileOverview overview = 1;
+}
+
+// Represents a the response from an UpdateOverviewRequest - intentionally left empty
+message UpdateOverviewResponse {}
+
+// Represents a specific configuration version associated with an instance
+message ConfigVersion {
+    // the instance identifier
+    string instance_id = 1;
+    // the version of the configuration
+    string version = 2;
+}
+
+// Represents a collection of files
+message FileOverview {
+    // A list of files
+    repeated File files = 1;
+    // the configuration version of the current set of files
+    ConfigVersion version = 2;
+}
+
+// Represents meta data about a file
+message File {
+    // Meta information about the file, the name (including path) and hash
+    FileMeta file_meta = 1;
+    // Action enumeration
+    enum FileAction {
+        // Default value, no action
+        FILE_ACTION_UNSPECIFIED = 0;
+        // No changes to the file
+        FILE_ACTION_UNCHANGED = 1;
```

Comment:
can probably remove UNCHANGED, and use UNSPECIFIED as the default, indicating no change.

Reply:
UNSPECIFIED is for an unspecified file FileAction. UNCHANGED is an explicit no file has changed. They have different purposes